plugins/opensslcryptoplugin/evp/evp.c

```diff
@@ -336,6 +336,108 @@ int osslevp_aes_auth_encrypt_plaintext(const EVP_CIPHER *evp_cipher,
     return ciphertext_length;
 }
 
+int osslevp_aes_auth_decrypt_ciphertext(const EVP_CIPHER *evp_cipher,
+                                        const unsigned char *init_vector,
+                                        const unsigned char *key,
+                                        int key_length,
+                                        const unsigned char *auth,
+                                        int auth_length,
+                                        unsigned char *tag,
+                                        int tag_length,
+                                        const unsigned char *ciphertext,
+                                        int ciphertext_length,
+                                        unsigned char **decrypted)
+{
+    int plaintext_length = 0;
+    int update_length = 0;
+    int final_length = 0;
+    unsigned char *plaintext = NULL;
+
+    if (evp_cipher == NULL || ciphertext_length <= 0 || ciphertext == NULL
+            || auth == NULL || auth_length <= 0 || tag == NULL || tag_length <= 0
+            || key_length <= 0 || key == NULL || decrypted == NULL) {
+        /* Invalid arguments */
+        fprintf(stderr,
+                "%s: %s\n",
+                "osslevp_aes_auth_decrypt_ciphertext()",
+                "invalid arguments, aborting decryption");
+        return -1;
+    }
+
+    /* Allocate the buffer for the decrypted output */
+    plaintext = (unsigned char *)malloc(ciphertext_length + AES_BLOCK_SIZE);
```

Comment:
Shouldn't the allocation happen after you know the actual decrypted size from `EVP_DecryptFinal_ex`?

Reply:
Hm, then how would you pass an allocated buffer to EVP_DecryptUpdate() to get the decrypted result?